core/templates/dev/head/pages/state_editor/state_properties/StateEditorService.js

```diff
@@ -78,7 +78,7 @@ oppia.factory('StateEditorService', [
         interaction.setHints(hints);
       },
       getInteraction: function() {
-        return interaction;
+        return angular.copy(interaction);
```

Comment:
Why do we do this here but not for other things like getMisconceptions etc.?

Just concerned that this might break something.

Reply:
This was actually changed to prevent the base interaction from being modified until QuestionUpdateService is called. It was used in solution validation service to check solution validity every time answer groups were changed, but this caused the base interaction being modified before it was 'officially' modified in QuestionUpdateService. Having it be copied explicitly doesn't affect the solution service.

Reply:
OK, sounds good. Should we do something similar for the other properties, though? I am concerned that a dev adding a new property to this file will not know which convention to follow.

Reply:
The only other similar function here seems to be `getMisconceptions()`, but in the question editor, misconceptions are read-only and never edited, hence it doesn't matter either way. 
I'll add the copy to that also for consistency.

Reply:
Done